[Feature][JsonSplit-api] modify API to Restful-01

marin-man opened the pull request:
#5498
This table records the modifications completed.

|name|API|UI||name|API|UI|
|:--:|:--:|:-:|:--:|:--:|:--:|:--:|
|ProcessDefinition| [1](https://github.com/apache/dolphinscheduler/pull/6016) | [1](https://github.com/apache/dolphinscheduler/pull/6016) ||AlertPluginInstance| [2](https://github.com/apache/dolphinscheduler/pull/6090) | [2](https://github.com/apache/dolphinscheduler/pull/6090) |
|ProcessInstances| [1](https://github.com/apache/dolphinscheduler/pull/6016) | [1](https://github.com/apache/dolphinscheduler/pull/6016) || DataSource|[2](https://github.com/apache/dolphinscheduler/pull/6090) | [2](https://github.com/apache/dolphinscheduler/pull/6090) |
|TaskDefinition| [1](https://github.com/apache/dolphinscheduler/pull/6016) | [1](https://github.com/apache/dolphinscheduler/pull/6016) ||Monitor| [3](https://github.com/apache/dolphinscheduler/pull/6100) |[3](https://github.com/apache/dolphinscheduler/pull/6100) |
|TaskInstances| [1](https://github.com/apache/dolphinscheduler/pull/6016) | [1](https://github.com/apache/dolphinscheduler/pull/6016) ||Queue| [3](https://github.com/apache/dolphinscheduler/pull/6100)  | [3](https://github.com/apache/dolphinscheduler/pull/6100) |
|Project| [1](https://github.com/apache/dolphinscheduler/pull/6016) | [1](https://github.com/apache/dolphinscheduler/pull/6016) ||Tenant| [3](https://github.com/apache/dolphinscheduler/pull/6100) | [3](https://github.com/apache/dolphinscheduler/pull/6100)  |
|AccessToken| [2](https://github.com/apache/dolphinscheduler/pull/6090) | [2](https://github.com/apache/dolphinscheduler/pull/6090) ||Scheduler| [3](https://github.com/apache/dolphinscheduler/pull/6100) | [3](https://github.com/apache/dolphinscheduler/pull/6100)  |
|AlertGroup| [2](https://github.com/apache/dolphinscheduler/pull/6090) | [2](https://github.com/apache/dolphinscheduler/pull/6090) ||Resource|4|4|
|WorkerGroup|4|4||UiPlugin|4|4|

[a document about modification](https://kdocs.cn/l/spOjvEzstRLk)

Commit:
1. update ProcessDefinition API
2. update TaskDefinition API

Commit:
1. update ProcessDefinition API
2. update TaskDefinition API

brave-lee commented on dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessDefinitionController.java:
```diff
@@ -110,7 +102,7 @@
         @ApiImplicitParam(name = "locations", value = "PROCESS_DEFINITION_LOCATIONS", required = true, type = "String"),
         @ApiImplicitParam(name = "description", value = "PROCESS_DEFINITION_DESC", required = false, type = "String")
     })
-    @PostMapping(value = "/save")
```
Should this be kept ?

marin-man replied:
In the standard document, we use POST to represent create.

brave-lee commented on dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessDefinitionController.java:
```diff
@@ -138,12 +130,12 @@ public Result createProcessDefinition(@ApiIgnore @RequestAttribute(value = Const
      * @return copy result code
      */
-    @ApiOperation(value = "copy", notes = "COPY_PROCESS_DEFINITION_NOTES")
+    @ApiOperation(value = "batchCopyProcessDefinitions", notes = "COPY_PROCESS_DEFINITION_NOTES")
```
please check

marin-man replied:
I have put it another way.
`@ApiOperation(value = "batchMoveByCodes", notes = "MOVE_PROCESS_DEFINITION_NOTES")`

marin-man committed:
Express batch copy in another way

ruanwenjun commented on dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessDefinitionController.java:
```diff
@@ -58,15 +58,7 @@
-import org.springframework.web.bind.annotation.ResponseBody;
-import org.springframework.web.bind.annotation.ResponseStatus;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
```
It is better not do this change, use import * is not a good practice.

ruanwenjun committed:
Update dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessDefinitionController.java

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestAttribute;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.bind.annotation.RestController;

marin-man reviewed (commented):
These also need to import other packages, such as PutMapping and DeleteMapping.  I didn't change this part. The IDEA editor automatically changes to *.

manmandm commented:
> These also need to import other packages, such as PutMapping and DeleteMapping. I didn't change this part. The IDEA editor automatically changes to *.

@ruanwenjun

manmandm commented:
> These also need to import other packages, such as PutMapping and DeleteMapping. I didn't change this part. The IDEA editor automatically changes to *.

wen-hemin commented on dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/ProcessDefinitionController.java:
```diff
@@ -81,7 +72,7 @@
 @Api(tags = "PROCESS_DEFINITION_TAG")
 @RestController
-@RequestMapping("projects/{projectCode}/process")
+@RequestMapping("projects/{projectCode}/process-definitions")
```
English singular is recommended.
project/{projectCode}/process-definition

marin-man replied:
ok

wen-hemin commented on dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/TaskDefinitionController.java:
```diff
@@ -160,14 +153,14 @@ public Result queryTaskDefinitionVersions(@ApiIgnore @RequestAttribute(value = C
         @ApiImplicitParam(name = "version", value = "VERSION", required = true, dataType = "Int", example = "100")
     })
-    @GetMapping(value = "/version/switch")
+    @GetMapping(value = "/{code}/version/switch/{version}")
```
Get or Post ?

Get:
/{code}/version/{version}

Post:
/{code}/version/{version}/switch

marin-man replied:
I think the first is good

gabry-lab replied:
/project/{code}/version/{version}/switch better

wen-hemin reviewed (commented):
Please check

marin-man committed:
modify definition statement

marin-man committed:
modify switch version statement

marin-man committed:
modify definition statement

marin-man committed:
Update ProcessDefinitionController.java

marin-man committed:
Update ProcessDefinitionController.java

marin-man committed:
Update TaskDefinitionController.java

gabry-lab reviewed (changes requested):
please add the API Excel doc as an attachment

Commit:
update process-definition, process-instances, task-definition, task-instances, project and their ui.

Commit:
Merge branch 'json_split_two' of https://github.com/manmandm/dolphinscheduler

Commit:
modify ui error

Commit:
modify ui error

Commit:
Merge branch 'json_split_two' of https://github.com/manmandm/dolphinscheduler

marin-man committed:
Merge branch 'json_split_two' into json_split_two

Commit:
修改导包方式

Commit:
Merge branch 'json_split_two' of https://github.com/manmandm/dolphinscheduler

Commit:
merge

marin-man committed:
Merge branch 'json_split_two' into json_split_two

marin-man committed:
Delete TaskDefinitionService.java

merge

Commit:
merge

marin-man committed:
Update TaskDefinitionService.java

merge

marin-man committed:
Update TaskDefinitionService.java

marin-man committed:
Update TaskDefinitionService.java

merge

gabry-lab reviewed (approved):
+1

marin-man committed:
modify import

Commit:
modify import

brave-lee reviewed (commented):
+1

brave-lee commented:
pls fix confilct first

marin-man committed:
Merge branch 'json_split_two' into json_split_two

marin-man committed:
Update actions.js

lenboo reviewed (approved):
+1
